scarlet/initialisation.py

```diff
@@ -0,0 +1,136 @@
+from .source import *
+from .interpolation import interpolate_observation
+from .observation import Observation
+from .wavelet import Starlet
+
+def build_initialisation_coadd(observations, wavelet = False):
+    """Build a channel weighted coadd to use for source detection
+
+    Parameters
+    ----------
+    observations: `~scarlet.observation.Observation`
+        Observation to use for the coadd.
+
+    Returns
+    -------
+    detect: array
+        2D image created by weighting all of the channels by SED
+    bg_cutoff: float
+        The minimum value in `detect` to include in detection.
+    """
+    try:
+        iter(observations)
+    except TypeError:
+        observations = [observations]
+    # The observation that lives in the same plane as the frame
+    loc = np.where([type(obs) is Observation for obs in observations])
+    # If more than one element is an `Observation`, then pick the first one as a reference (arbitrary)
+    obs_ref = observations[loc[0][0]]
+
+    n_images = 0
+    coadd = 0
+    jacobian = 0
+    weights = 0
+    for obs in observations:
+        try:
+            weights = np.array([w[w > 0].mean() for w in obs.weights])
+        except:
+            raise AttributeError(
+                "Observation.weights missing! Please set inverse variance weights"
+            )
+
+        if obs is obs_ref:
+            if wavelet is True:
+                images = Starlet(obs.images).filter()
+            else:
+                images = obs.images
+        else:
+            #interpolate low-res to reference resolution
+            images = interpolate_observation(obs, obs_ref.frame, wave_filter = wavelet)
+        n_images += len(images)
+        if wavelet is True:
```

Comment:
I'm not sure I like this as the parameter name to decide how to build the coadd. I would prefer to either make separate functions for different coadd types or use a more descriptive name for the different coadd types and pass that as either a "magic string" or (as I prefer) and enum.

Reply:
I might actually drop this all together. It doesn't bring anything. The results are pretty much the same.

Reply:
Name chaged to `filter_coadd`